pkg/controller/nodemaintenance/check_lease_update.go

```diff
@@ -0,0 +1,431 @@
+}
+
+func getMockNode() (*corev1.Node) {
+	node := &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "miau",
+					UID:  "foobar",
+				},
+	}
+	return node
+}
+
+var _ = Describe("checkLeaseUpdate", func() {
+
+	DescribeTable("check lease supported",
+		func(initLease *coordv1beta1.Lease, leasePostcondition *coordv1beta1.Lease) {
+			node := getMockNode()
+			objs := []runtime.Object{
+				initLease,
+			}
+			cl  := fake.NewFakeClient(objs...)
+			timeNow :=  time.Unix(NowTime,0)
+
+			name := apitypes.NamespacedName{Namespace: LeaseNamespace, Name: node.Name}
+			leaseIn := &coordv1beta1.Lease{}
+			err := cl.Get(context.TODO(), name, leaseIn)
+			Expect(err).NotTo(HaveOccurred())
+
+			leaseRet, err, failedUpdateOwnedLease := updateLease(cl, node, leaseIn, timeNow, LeaseDurationInSeconds)
+			if leasePostcondition == nil {
+				Expect(err).To(HaveOccurred())
+
+				var nilRef *coordv1beta1.Lease
+				Expect(leaseRet).To(Equal(nilRef))
+			} else {
+				Expect(err).NotTo(HaveOccurred())
+
+				Expect(failedUpdateOwnedLease).To(Equal(false))
+				lease := &coordv1beta1.Lease{}
+				err = cl.Get(context.TODO(), name, lease)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(leaseRet).To(Equal(lease))
+
+				Expect(len(lease.ObjectMeta.OwnerReferences)).To(Equal(1))
+				Expect(len(leasePostcondition.ObjectMeta.OwnerReferences)).To(Equal(1))
+
+				leaseRef := lease.ObjectMeta.OwnerReferences[0]
+				leasePost := leasePostcondition.ObjectMeta.OwnerReferences[0]
+
+				Expect(leasePost.APIVersion).To(Equal(leaseRef.APIVersion))
+				Expect(leasePost.Kind).To(Equal(leaseRef.Kind))
+				Expect(leasePost.Name).To(Equal(leaseRef.Name))
+				Expect(leasePost.UID).To(Equal(leaseRef.UID))
+
+				Expect(compareString(lease.Spec.HolderIdentity, leasePostcondition.Spec.HolderIdentity)).To(Equal(true))
+				Expect(compareTime(lease.Spec.RenewTime, leasePostcondition.Spec.RenewTime)).To(Equal(true))
+				Expect(compareTime(lease.Spec.AcquireTime, leasePostcondition.Spec.AcquireTime)).To(Equal(true))
+				Expect(compareInt32(lease.Spec.LeaseDurationSeconds, leasePostcondition.Spec.LeaseDurationSeconds)).To(Equal(true))
+
+				Expect(compareInt32(lease.Spec.LeaseTransitions, leasePostcondition.Spec.LeaseTransitions)).To(Equal(true))
+			}
+		},
+
+
+		Entry("fail to update valid lease with different holder identity",
+				&coordv1beta1.Lease{
+					ObjectMeta :metav1.ObjectMeta{
+						Name:	getMockNode().Name,
+						Namespace: LeaseNamespace,
+						OwnerReferences: []metav1.OwnerReference{
+							metav1.OwnerReference{
+								APIVersion: "v1",
+								Kind:       "Node",
+								Name:       "@",
+								UID:        "#",
+							},
+						},
+					},
+					Spec: coordv1beta1.LeaseSpec{
+						HolderIdentity:       makeString("miau"),
+						LeaseDurationSeconds: makeInt32(3200),
+						AcquireTime:          nil,
+						RenewTime:            getTimeLease(int64(NowTime-1)),
+						LeaseTransitions:     nil,
+					},
+				},
+				nil,
```

Comment:
Shouldn't there be an error to check for too?

Reply:
done, checking the error as well,